.github/workflows/ci.yml

```diff
@@ -26,8 +26,39 @@ jobs:
       AWS_SECRET_ACCESS_KEY: "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
       LOG_PATH: /tmp/debug-logs
       PIP_LOG_FILE: /tmp/pip.log
+      HOMEBREW_NO_AUTO_UPDATE: 1
     steps:
       - uses: actions/checkout@v3
+      - name: Update Homebrew and save docker version
+        if: runner.os == 'macOS'
+        run: |
+          brew update --preinstall
+          cat "$(brew --repository)/Library/Taps/homebrew/homebrew-core/Formula/docker.rb" > .github/brew-formulae
+      - name: Configure Homebrew docker cache files
+        uses: actions/cache@v3
+        if: runner.os == 'macOS'
+        with:
+          path: |
+            ~/Library/Caches/Homebrew/docker--*
+            ~/Library/Caches/Homebrew/downloads/*--docker-*
+          key: brew-${{ hashFiles('.github/brew-formulae') }}
+          restore-keys: brew-
+      - name: Install Docker if on MacOS and start colima
+        id: install_mac_docker
+        if: runner.os == 'macOS'
+        run: |
+          brew install docker
+          # Docker engine is no longer available because of licensing
+          # Alternative Colima is part of the github macOS runner
+          # SAM v1.47.0+ needed for colima support, unable to use Python 3.6
+          colima start
+          # Ensure colima is configured for later user
+          echo "DOCKER_HOST=unix://$HOME/.colima/default/docker.sock" >> $GITHUB_ENV
+          # Verify Docker
+          docker ps
```

Comment:
Hows this verify docker?  Seems like extra and for logging purposes?

Reply:
I wanted to do a docker CLI to server connection check.
`docker info`
`docker version`

would also be options that fail if docker CLI cannot connect with the backend.  Though those are more verbose.